backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -361,17 +363,10 @@ public AddTestResultResponse addMultiplexResult(
 
     boolean deliveryStatus =
         deliveryStatuses.isEmpty() || deliveryStatuses.stream().anyMatch(status -> status);
-    return new AddTestResultResponse(savedOrder, deliveryStatus);
-  }
-
-  private void reportTestEventToRS(TestEvent savedEvent) {
-    if (savedEvent.hasCovidResult()) {
-      _testEventReportingService.report(savedEvent);
-    }
 
-    if (savedEvent.hasFluResult() && fhirReportingEnabled) {
-      _fhirQueueReportingService.report(savedEvent);
-    }
+    AddTestResultResponse response = new AddTestResultResponse(savedOrder, deliveryStatus);
+    applicationEventPublisher.publishEvent(new ReportTestEventToRSEvent(savedOrder.getTestEvent()));
+    return response;
```

Comment:
do we need the `response` variable? can we just return without declaring it?

Reply:
Oh good catch I forgot to clean that one up when refactoring it